### PR TITLE
Add additional icons in recent events

### LINKED ISSUE #8410: Add an icon statements for events in "recent"
For example, it may have this:

image

This is necessary to eliminate the "voids" in this graph

### PR DESCRIPTION
**Pull request by KeziahMoselle**
## Events

- [x] beatmapPlaycount
- [x] beatmapsetApprove
- [x] beatmapsetDelete
- [x] beatmapsetRevive
- [x] beatmapsetUpdate
- [x] beatmapsetUpload
- [x] rankLost
- [x] userSupportAgain
- [x] userSupportFirst
- [x] userSupportGift
- [x] usernameChange

### Colors

Should we color them as proposed initially? If so how?
With additionnal classes based on event type in `resources/assets/less/bem/page-extra.less` ?

closes #8410

### CODE REVIEW

**Comment by peppy** [comment]
Are you able to provide a screenshot of how this looks so far?

**Comment by KeziahMoselle** [comment]
Not yet, I need to seed the user profile with all possible types of events but I didn't see how yet

**Comment by Loreos7** [comment]
Perhaps this down below would sute better "has been revived from eternal slumber"
image
Although it could be smth like osu circle with the wings

And dont forget about the colors, because, at least, there are several statements for the beatmap: ranked, qualified, loved, approved

**Comment by KeziahMoselle** [comment]
> Perhaps this down below would sute better for "has been revived from eternal slumber" image Although it could be smth like osu circle with the wings
> 
> And dont forget about the colors, because, at least, there are several statements for the beatmap: ranked, qualified, loved, approved

afaik the icon wave pulse isn't available in font awesome v5

**Comment by RockRoller01** [comment]
I believe a play icon for "map has been played x times" would be better, not to mention thats also what the play count medal uses

https://fontawesome.com/icons/play?s=solid

File: resources/assets/less/bem/profile-extra-entries.less [hunk 1]
```diff
@@ -94,6 +94,38 @@
...
+      color: @green;
+    }
+
+    &--yellow {
+      color: @yellow;
+    }
+
+    &--pink {
+      color: @pink;
+    }
```
Comment (KeziahMoselle) [inline resources/assets/less/bem/profile-extra-entries.less:128]: I am unsure about this class organization 🤔

**Comment by KeziahMoselle** [comment]
Here is a first version with colors: 
image

File: resources/assets/lib/profile-page/parse-event.tsx [hunk 2]
```diff
@@ -60,6 +64,7 @@ export default function parseEvent(event: EventJson, modifiers: Modifiers): { ba
 
     case 'beatmapsetUpdate':
       return {
+        badge: <span className='fas fa-sync-alt profile-extra-entries__icon--green' />,
```
Comment (nanaya) [inline resources/assets/lib/profile-page/parse-event.tsx:67]:
```suggestion
        badge: <span className='fas fa-sync-alt profile-extra-entries__icon--green' />,
        iconModifiers: 'green',
```

And then adjust the render function to use the modifiers and apply it correctly to the `__icon` element.

File: resources/assets/less/bem/profile-extra-entries.less [hunk 3]
```diff
@@ -94,6 +94,38 @@
...
+    &--approved {
+      color: var(--beatmapset-approved-bg);
+    }
+
+    &--danger {
+      color: hsl(var(--hsl-red-3));
+    }
+
+    &--green {
+      color: @green;
```
Comment (nanaya) [inline resources/assets/less/bem/profile-extra-entries.less:119]: those `@$colour`s shouldn't be used anymore. Use `--hsl-$colour-$ver` instead.

**Review by nanaya — CHANGES_REQUESTED** [review]

**Comment by KeziahMoselle** [comment]
Updated @nanaya 👍

**Review by nanaya — APPROVED** [review]

**Pull request merged**